Fix disabled unit tests

roelandkulk opened the pull request.

MarcelWildenburg commented on src/test/java/org/littil/api/exception/ThrowableMapperTest.java:
```diff
@@ -16,13 +18,16 @@
 
 @QuarkusTest
 @TestHTTPEndpoint(GuestTeacherResource.class)
```
@pschildkamp : should we use **@QuarkusTestResource(APIManagementMock.class)** here ?

MarcelWildenburg commented on src/test/java/org/littil/api/guestTeacher/service/GuestTeacherServiceTest.java:
```diff
@@ -147,32 +150,27 @@ void givenSaveTeacherUnknownErrorOccurred_thenShouldThrowPersistenceException()
         final GuestTeacher expectedGuestTeacher = new GuestTeacher();
         expectedGuestTeacher.setId(entity.getId());
         expectedGuestTeacher.setSurname(entity.getSurname());
         expectedGuestTeacher.setFirstName(entity.getFirstName());
```
ExpectedQuestTeacher is not used, so these lines can be removed.

roelandkulk replied:
Thanks, this has been removed now.

MarcelWildenburg reviewed (commented):
I have added some comments.
The saveTeacher method is only tested for PersistenceException. 
Maybey you can add a test for 
- Happy flow
- saveTeacher for unknown user

roelandkulk commented:
> I have added some comments. The saveTeacher method is only tested for PersistenceException. Maybey you can add a test for
> 
>     * Happy flow
> 
>     * saveTeacher for unknown user

These have been added.

roelandkulk commented:
I have refactored GuestTeacherResourceTest.
Only givenDeleteTeacherById_thenShouldDeleteSuccessfully() is not working. The delete request is caught by AbstractSecurityInterceptor, tokenHelper.getCustomClaim() returns null. Unfortunately it's not possible to mock this.

I have removed @NotEmpty(message = "{GuestTeacher.locale.required}") from GuestTeacherPostResource and GuestTeacher since providing the locale is optional. There is still some work needed to map locale to location.country and back when a locale is provided but I think this is more suitable under a new issue / branch.
